### ISSUE TITLE
Proposal: split parsers as separate Python package 

### ISSUE THREAD

**Issue by pauliacomi** [AIF-development-team/adsorptioninformationformat#27]
As the title says, I propose to have a separate package on `pypi` that would act as a collection of python parsers.

# Background

I am also maintaining a collection of parsers in pyGAPS. I have recently added a few tweaks for better parsing of Micromeritics files and for reading of reports from 3P instruments. It would be ideal if I could contribute these to the parsing used by the AIF. It would also be good if I could benefit from changes in this repo (like the recent JIS tweaks). However, I find it is difficult as the codebases have diverged, it takes a lot of time to compare and correctly integrate changes from one to another.

# Proposition

Common parsing code can be placed in a separate repo, under `AIF-dev-team`. This would then be published to ``pypy`` as an independent package, for ex `adsorption-file-parsers`. Anyone wishing to use them in their projects could then install and depend on this package, allowing for a single source of truth. Any improvements could also be contributed directly here and propagate seamlessly.

Would be open to maintaining the package.

# Implementation

For each type of file, a module would be created in the parsing package. The module would have a main function that deals with files of the correct type, perhaps with other options.

```
from adsorption_file_parsers import parse_micromeritics_rep

file_data = parse_micromeritics_rep("path/to/file", *options)
```

# Challenges

* Should depend on least number of external packages.
* A common return type should be agreed upon, ideally pure Python, e.g.:
```
data = {
    "metadata": {}, # dict
    "pointdata": {}, # dataframe? dict? lists?
}
```

**Comment by jackevansadl** [comment]
I agree with this proposal and will add you to the development team.
There is a great advantage to separating this work as we have all put a lot of effort into these parsers and it would be great for them to be made available to other routines.
It may take a few versions to push these changes to the raw2aif routine but I definitely agree it is the best step forward.

**Comment by pauliacomi** [comment]
Great, thanks Jack.
As of now I suggest that I'll create a skeleton package which integrates the current code and tests.
We can then further the discussion on that repo with exact implementation details, best practices and integration with the codebases going forward.
